clang-tools-extra/clangd/refactor/Rename.cpp

```diff
@@ -681,12 +957,22 @@ renameOutsideFile(const NamedDecl &RenameDecl, llvm::StringRef MainFilePath,
            ExpBuffer.getError().message());
       continue;
     }
+    std::string RenameIdentifier = RenameDecl.getNameAsString();
+    std::optional<Selector> Selector = std::nullopt;
+    llvm::SmallVector<llvm::StringRef, 8> NewNames;
+    if (const auto *MD = dyn_cast<ObjCMethodDecl>(&RenameDecl)) {
```

Comment:
A high-level comment. Instead of dealing with a vector of `StringRef` for `NewNames`, I think we should use `SymbolName`. It is already designed to be able to represent Objective-C selectors and that way you could do the name splitting in `SymbolName`.

Reply:
i'd rather leave that to a follow up change so that we can close the loop here, if that's OK.